arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,644 @@
+      video_reader->open( config->get_value<std::string>("video_source") );
+      if (video_reader->get_implementation_capabilities()
+        .has_capability(video_input::HAS_METADATA))
+      {
+        sfm_constraint_ptr->set_metadata(video_reader->metadata_map());
+        kwiver::vital::timestamp ts;
+        video_reader->next_frame( ts );
+        first_frame = video_reader->frame_image();
+      }
+      else
+      {
+        return;
+      }
+    }
+    else
+    {
+      return;
+    }
+
+    using kwiver::vital::simple_camera_intrinsics;
+    using kwiver::vital::simple_camera_perspective;
+    using kwiver::vital::frame_id_t;
+    using kwiver::vital::metadata_sptr;
+    using kwiver::vital::intrinsics_from_metadata;
+    using kwiver::vital::local_geo_cs;
+
+    #define GET_K_CONFIG(type, name) \
+    config->get_value<type>(bc + #name, K_def.name())
+
+    simple_camera_intrinsics K_def;
+    const std::string bc = "video_reader:base_camera:";
+    auto K = std::make_shared<simple_camera_intrinsics>(
+      GET_K_CONFIG(double, focal_length),
+      GET_K_CONFIG(kwiver::vital::vector_2d, principal_point),
+      GET_K_CONFIG(double, aspect_ratio),
+      GET_K_CONFIG(double, skew));
+
+    auto baseCamera = simple_camera_perspective();
+    baseCamera.set_intrinsics(K);
+    auto md = sfm_constraint_ptr->get_metadata()->metadata();
+    kwiver::vital::camera_map::map_camera_t camMap;
+    if (!md.empty())
+    {
+      std::map<frame_id_t, metadata_sptr> mdMap;
+
+      for (auto const& mdIter : md)
+      {
+        // TODO: just using first element of metadata vector for now
+        mdMap[mdIter.first] = mdIter.second[0];
+      }
+
+      bool init_cams_with_metadata =
+        config->get_value<bool>("initialize_cameras_with_metadata", true);
+
+      if (init_cams_with_metadata)
+      {
+        auto im = first_frame;
+        K->set_image_width(static_cast<unsigned>(im->width()));
+        K->set_image_height(static_cast<unsigned>(im->height()));
+        baseCamera.set_intrinsics(K);
+
+        bool init_intrinsics_with_metadata =
+        config->get_value<bool>("initialize_intrinsics_with_metadata", true);
+        if (init_intrinsics_with_metadata)
+        {
+          // find the first metadata that gives valid intrinsics
+          // and put this in baseCamera as a backup for when
+          // a particular metadata packet is missing data
+          for (auto mdp : mdMap)
+          {
+            auto md_K =
+              intrinsics_from_metadata(*mdp.second,
+                                       static_cast<unsigned>(im->width()),
+                                       static_cast<unsigned>(im->height()));
+            if (md_K != nullptr)
+            {
+              baseCamera.set_intrinsics(md_K);
+              break;
+            }
+          }
+        }
+
+        local_geo_cs lgcs = sfm_constraint_ptr->get_local_geo_cs();
+        camMap =
```

Comment:
camMap as camera_map_sptr instead of camera_map_t does not compile.  Granted, since it is not used right now, I do not need to keep the return from the function.

Reply:
Ah, we may need to package that camera map into a sptr.  Something like 
```c++
camera_map_ptr = std::make_shared<simple_camera_map>(camMap);
```
syntax may be slight off, I didn't test this.